dashboard/cron.py

```diff
@@ -128,25 +128,29 @@ def update_user(self):
         # loop through multiple course ids
         for data_warehouse_course_id in Course.objects.get_supported_courses():
 
-            # select all student registered for the course
-            user_sql = f"""select u.name AS name,
-                        p.sis_user_id AS sis_id,
-                        p.unique_name AS sis_name,
-                        u.global_canvas_id AS id,
-                        c.current_score AS current_grade,
-                        c.final_score AS final_grade,
-                        '{data_warehouse_course_id}' as course_id
-                        from user_dim u,
-                        pseudonym_dim p,
-                        course_score_fact c,
-                        (select e.user_id as user_id, e.id as enrollment_id from enrollment_dim e
-                        where e.course_id = '{data_warehouse_course_id}'
-                        and e.type='StudentEnrollment'
-                        and e.workflow_state='active' ) as e
-                        where p.user_id=u.id
-                        and u.id = e.user_id
-                        and c.enrollment_id =  e.enrollment_id
-                        and p.sis_user_id is not null
+            # select all student, tas, and instructors registered for the course
+            user_sql = f"""
+                        with enrollment as (
+                            select DISTINCT e.user_id, e.type, c.current_score, c.final_score
+                            from enrollment_dim e
+                            left join course_score_fact c ON c.enrollment_id = e.id
+                            where e.course_id = '{data_warehouse_course_id}'
+                            and e.workflow_state = 'active'
+                            and e.type in ('StudentEnrollment', 'TaEnrollment', 'TeacherEnrollment')
+                        )
+                        select u.name as name,
+                               p.sis_user_id as sis_id,
+                               p.unique_name as sis_name,
+                               u.global_canvas_id as id,
+                               e.current_score as current_grade,
+                               e.final_score as final_grade,
+                               e.type as enrollment_type,
+                               '{data_warehouse_course_id}' as course_id
+                        from user_dim u
+                        join pseudonym_dim p ON p.user_id = u.id
+                        join enrollment e ON u.id = e.user_id
+                        where p.workflow_state = 'active'
+                        and p.authentication_provider_id = '{settings.CANVAS_PRIMARY_AUTH_PROVIDER_ID}'
```

Comment:
This value authentication_provider_id looks to be "null" for all of our values in the table. Joining this this against pseudonym_dim would have to be changed to allow for "is null" or be an optional condition.